swiglpk/glpk.i

```diff
@@ -72,9 +72,14 @@ PyObject* get_col_primals(glp_prob *P) {
     int n = glp_get_num_cols(P);
     PyObject* list = PyList_New(n);
     double prim = 0.0;
-    int i=0;
+    int n_int = glp_get_num_int(P);
+    int i = 0;
     for(i=1; i<=n; i++){
-        prim = glp_get_col_prim(P, i);
+        if (n_int == 0) {
```

Comment:
Since speed is a concern, wouldn't it be better to perform the conditional check outside the loop? Or is this anyway optimized by the compiler?